tests/infra/rates.py

```diff
@@ -21,19 +21,34 @@ def __init__(self, primary):
         self.commit = 0
 
     def __str__(self):
-        out_list = ["----------- tx rates -----------"]
-        out_list.append("----- mean ----: " + str(mean(self.tx_rates_data)))
-        out_list.append(
-            "----- harmonic mean ----: " + str(harmonic_mean(self.tx_rates_data))
-        )
-        out_list.append(
-            "---- standard deviation ----: " + str(pstdev(self.tx_rates_data))
-        )
-        out_list.append("----- median ----: " + str(median(self.tx_rates_data)))
-        out_list.append("---- max ----: " + str(max(self.tx_rates_data)))
-        out_list.append("---- min ----: " + str(min(self.tx_rates_data)))
-        out_list.append("----------- tx rates histogram -----------")
-        out_list.append(json.dumps(self.histogram_data, indent=4))
+        out_list = []
+
+        def format_title(s):
+            out_list.append(f"{s:-^42}")
+
+        def format(s, n):
+            out_list.append(f"--- {s:>20}: {n:>12.2f} ---")
+
+        format_title("Summary")
+        format("mean", mean(self.tx_rates_data))
+        format("harmonic mean", harmonic_mean(self.tx_rates_data))
+        format("standard deviation", pstdev(self.tx_rates_data))
+        format("median", median(self.tx_rates_data))
+        format("max", max(self.tx_rates_data))
+        format("min", min(self.tx_rates_data))
+
+        format_title("Histogram")
+        buckets_list = self.histogram_data["buckets"]
+        buckets = {tuple(e[0]): e[1] for e in buckets_list}
```

Comment:
tuples of one?

Reply:
e[0] is a 2-element list, converted to a tuple so it can be hashed as dict-key. The json format is a list of pairs (lists), where each pair's first element is a [begin, end] pair, and the second element is the count.

This conversion is:
```
[[[0, 5], 3], [[6, 10], 1], [[11, 15], 7]]
=>
{(0, 5): 3, (6, 10): 1, (11, 15): 7}
```